src/meshapi/validation.py

```diff
@@ -102,17 +107,29 @@ def __init__(self, street_address: str, city: str, state: str, zip: int):
 
         # Now that we have the bin, we can definitively get the height from
         # NYC OpenData
-        query_params = {
-            "$where": f"bin={self.bin}",
-            "$select": "heightroof,groundelev",
-            "$limit": 1,
-        }
-        nyc_dataset_req = requests.get(f"https://data.cityofnewyork.us/resource/qb5r-6dgf.json", params=query_params)
-        nyc_dataset_resp = json.loads(nyc_dataset_req.content.decode("utf-8"))
-
-        if len(nyc_dataset_resp) == 0:
-            raise AddressAPIError(
+        try:
+            query_params = {
+                "$where": f"bin={self.bin}",
+                "$select": "heightroof,groundelev",
+                "$limit": 1,
+            }
+            nyc_dataset_req = requests.get(
+                "https://data.cityofnewyork.us/resource/qb5r-6dgf.json",
+                params=query_params,
+                timeout=DEFAULT_EXTERNAL_API_TIMEOUT_SECONDS,
+            )
+            nyc_dataset_resp = json.loads(nyc_dataset_req.content.decode("utf-8"))
+
+            if len(nyc_dataset_resp) == 0:
+                logging.warning(f"Empty response from nyc open data about altitude of ({self.bin})")
+                raise OpenDataAPIError
+            else:
+                self.altitude = float(nyc_dataset_resp[0]["heightroof"]) + float(nyc_dataset_resp[0]["groundelev"])
+        except OpenDataAPIError:
+            self.altitude = INVALID_ALTITUDE
+            logging.warning(
                 f"(NYC) DOB BIN ({self.bin}) not found in NYC OpenData while trying to query for altitude information"
             )
-
-        self.altitude = float(nyc_dataset_resp[0]["heightroof"]) + float(nyc_dataset_resp[0]["groundelev"])
+        except Exception as e:
+            self.altitude = INVALID_ALTITUDE
```

Comment:
Altitude is optional on the `Building` model, so we maybe should make this `None` here? I think that would be better than having to check for `None` _and_ `-1` everywhere that reads altitude values. Also, technically `-1` could be a valid altitude (for an extremely short building in an extremely flood prone area).

Reply:
Good call, changed the constant.